jax/_src/random.py

```diff
@@ -36,6 +36,15 @@
 from jax._src.util import prod
 
 
+IntOrFloat = Union[float, int, np.integer, np.floating]
+IntFloatOrArray = Union[jnp.ndarray, IntOrFloat]
+IntOrArray = Union[jnp.ndarray, int, np.integer]
```

Comment:
I used `IntFloatOrArray` and `IntOrArray` in many places, which was sometimes broader than the existing annotation.  I tested to make sure that the functions support it.